active/0002-centalized-mgmt-resource-types-provider-versions.md

```diff
@@ -0,0 +1,247 @@
+
+The example below shows a usage of the proposed syntax
+
+```bicep
+import az
+import kubernetes as k8s // uses optional aliasing of the
+```
+
+The resolution of the symbols will be resolved from the builtin `bicepconfig.json` under a new section:
+
+```json
+{
+  // prior bicepconfig.json sections
+  // ...
+
+  "providers": {
+    "az": {
+      // A package reference string (below) gestures Bicep to source the types from the registry address.
+      "source": "mcr.microsoft.com/bicep/providers/az:0.2.3", 
+      "isDefaultImport": true // Gestures Bicep to implicitly import the namespace identifier to the global scope
+    },
+    "kubernetes": {
+      // Using the value "builtin" (below) gestures Bicep to source types from the statically compiled NuGet package reference 
+      "source": "builtin", 
+      "isDefaultImport": false
+    }
+  }
+}
+```
+
+Since previous provider declaration syntax forms continue to be supported, we introduce the following constraints:
+
+- To ensure consistency with pre-existing handling of `bicepconfig.json` the configuration file closest to the Bicep file in the directory hierarchy is used. The [drawbacks](#drawbacks) of this constraint are described below.
+- The `sys` namespace is coupled to the Bicep bits version and cannot be overriden or dynamically uploaded using the mechanism described above
+
+#### Remove support for syntax `import '{providerName}@{providerVersion}' [as {optionalAlias}]`
+
+The syntax below currently in use to import built-in providers will become invalid and result in a diagnostic as a result of this change
+
+```bicep
+import 'sys@1.0.0'
+import 'kubernetes@1.0.0' as k8s
+```
+
+#### Change the delimiter '@' used by the supported provider declaration syntax to ':' to align with module reference syntax and increase consistency
+
+The syntax shown below uses ':' instead of '@' as a delimiter for provider declaration statements
+
+```bicep
+import 'br:mcr.microsoft.com/bicep/providers/az:0.2.3' // notice its NOT using az@0.2.3
+import 'br/public:az:0.2.3' as myAz
+```
+
+## Drawbacks
+
+> Discuss any potential drawbacks associated with this approach. Why might this _not_ be a suitable choice? Consider implementation costs, integration with existing and planned features, and the migration cost for Bicep users. Clearly identify if it constitutes a breaking change.
+
+1. The proposal uses `bicepconfig.json` to declare the provider information and is ruled by `bicepconfig.json` merge semantics. The merge semantics for `bicepconfig.json` specify that for a given Bicep source file, the configuration file closest to the Bicep file in the directory hierarchy is used. This behavior is not intuitive and may cause users that have specified multiple config files in their project file-structure to experience unexpected behaviors that may include build errors when Bicep attempts to resolve a namespace identifier. This is illustrated by the example below.
```

Comment:
I would consider this a separate discussion. We could adopt the proposed changes whether or not `bicepconfig.json` allows merges, and there's no reason this proposal should be gated on that work.

Reply:
agreed, I listed this item in the out of scope, but I did want to call out the drawback that results of this merge semantics not existing